packages/@pollyjs/adapter-node-http/src/index.js

```diff
@@ -63,16 +65,28 @@ export default class HttpAdapter extends Adapter {
       interceptor.intercept(/.*/, m).reply(function(_, body, respond) {
         const { req, method } = this;
         const { headers } = req;
-        const parsedArguments = parseRequestArguments(
+        const parsedArguments = normalizeClientRequestArgs(
           ...REQUEST_ARGUMENTS.get(req)
         );
         const url = getUrlFromOptions(parsedArguments.options);
-
-        // body will always be a string unless the content-type is application/json
-        // in which nock will then parse into an object. We have our own way of
-        // dealing with json content to convert it back to a string.
-        if (body && typeof body !== 'string') {
-          body = JSON.stringify(body);
+        const contentType = (headers['content-type'] || '').toString();
+        const isMultiPart = contentType.includes('multipart');
+
+        if (body) {
```

Comment:
Seems some overlap between these changes and https://github.com/Netflix/pollyjs/pull/257/files

Reply:
Yeah there is some overlap. @swashcap can validate and we can add some more test cases in that PR.